vault/core.go

```diff
@@ -2947,6 +2967,104 @@ type LicenseState struct {
 	Terminated bool
 }
 
+type InFlightRequests struct {
+	l                sync.RWMutex
```

Comment:
I'd like us to avoid having both a mutex and sync.Map; if we're going to use a mutex to guard the map, we may as well just use a regular map, since it's way cheaper than sync.Map if concurrent access is protected otherwise.

That said, I think you could get rid of the lock if you stored `InFlightReqData` instead of `*InflightReqData`.  The race occurs when we try to update data referenced by a pointer while another goroutine is trying to load the data.  If instead updating consists of copying the data out of the map, updating that copy, then storing a copy back into the map, I don't think it'll be racy.